bee-ledger/src/workers/consensus/worker.rs

```diff
@@ -94,23 +98,37 @@ where
         ));
     }
 
-    let mut metadata = WhiteFlagMetadata::new(milestone.essence().index(), milestone.essence().timestamp());
+    let mut metadata = WhiteFlagMetadata::new(
+        milestone.essence().index(),
+        milestone.essence().timestamp(),
+        Some(*milestone.essence().previous_milestone_id()),
+    );
 
     white_flag(tangle, storage, message.parents(), &mut metadata).await?;
 
-    if !metadata.merkle_proof.eq(&milestone.essence().merkle_proof()) {
-        return Err(Error::MerkleProofMismatch(
+    if !metadata
+        .confirmed_merkle_proof
+        .eq(&milestone.essence().confirmed_merkle_proof())
```

Comment:
```suggestion
    if metadata
        .confirmed_merkle_proof !=
        &milestone.essence().confirmed_merkle_proof()
```

Reply:
I won't commit your suggestion because of formatting but I'll do it.